web/includes/actions/filter.php

```diff
@@ -19,63 +19,64 @@
-          $sql .= ', AutoMove = 1, AutoMoveTo='. validInt($_REQUEST['filter']['AutoMoveTo']);
-        } else {
-          $sql .= ', AutoMove = 0'; 
-        }
-        $sql .= ', UpdateDiskSpace = '. ( !empty($_REQUEST['filter']['UpdateDiskSpace']) ? 1 : 0);
-        $sql .= ', Background = '. ( !empty($_REQUEST['filter']['Background']) ? 1 : 0);
-        $sql .= ', Concurrent  = '. ( !empty($_REQUEST['filter']['Concurrent']) ? 1 : 0);
+      $sql = '';
+      $_REQUEST['filter']['Query']['sort_field'] = validStr($_REQUEST['filter']['Query']['sort_field']);
+      $_REQUEST['filter']['Query']['sort_asc'] = validStr($_REQUEST['filter']['Query']['sort_asc']);
+      $_REQUEST['filter']['Query']['limit'] = validInt($_REQUEST['filter']['Query']['limit']);
+      if ( $action == 'execute' ) {
+        $tempFilterName = '_TempFilter'.time();
+        $sql .= ' Name = \''.$tempFilterName.'\'';
+      } else {
+        $sql .= ' Name = '.dbEscape($_REQUEST['filter']['Name']);
+      }
+      $sql .= ', Query = '.dbEscape(jsonEncode($_REQUEST['filter']['Query']));
+      $sql .= ', AutoArchive = '.(!empty($_REQUEST['filter']['AutoArchive']) ? 1 : 0);
+      $sql .= ', AutoVideo = '. ( !empty($_REQUEST['filter']['AutoVideo']) ? 1 : 0);
+      $sql .= ', AutoUpload = '. ( !empty($_REQUEST['filter']['AutoUpload']) ? 1 : 0);
+      $sql .= ', AutoEmail = '. ( !empty($_REQUEST['filter']['AutoEmail']) ? 1 : 0);
+      $sql .= ', AutoMessage = '. ( !empty($_REQUEST['filter']['AutoMessage']) ? 1 : 0);
+      $sql .= ', AutoExecute = '. ( !empty($_REQUEST['filter']['AutoExecute']) ? 1 : 0);
+      $sql .= ', AutoExecuteCmd = '.dbEscape($_REQUEST['filter']['AutoExecuteCmd']);
+      $sql .= ', AutoDelete = '. ( !empty($_REQUEST['filter']['AutoDelete']) ? 1 : 0);
+      if ( !empty($_REQUEST['filter']['AutoMove']) ? 1 : 0) {
+        $sql .= ', AutoMove = 1, AutoMoveTo='. validInt($_REQUEST['filter']['AutoMoveTo']);
+      } else {
+        $sql .= ', AutoMove = 0'; 
+      }
+      $sql .= ', UpdateDiskSpace = '. ( !empty($_REQUEST['filter']['UpdateDiskSpace']) ? 1 : 0);
+      $sql .= ', Background = '. ( !empty($_REQUEST['filter']['Background']) ? 1 : 0);
+      $sql .= ', Concurrent = '. ( !empty($_REQUEST['filter']['Concurrent']) ? 1 : 0);
 
-        if ( $_REQUEST['Id'] and ( $action == 'Save' ) ) {
-          dbQuery('UPDATE Filters SET ' . $sql. ' WHERE Id=?', array($_REQUEST['Id']));
-        } else {
-          dbQuery('INSERT INTO Filters SET' . $sql);
-          $_REQUEST['Id'] = dbInsertId();
-        }
-        if ( $action == 'execute' ) {
-          executeFilter( $tempFilterName );
-        }
+      if ( $_REQUEST['Id'] and ( $action == 'Save' ) ) {
+        dbQuery('UPDATE Filters SET '.$sql.' WHERE Id=?', array($_REQUEST['Id']));
+      } else {
+        dbQuery('INSERT INTO Filters SET'.$sql);
+        $_REQUEST['Id'] = dbInsertId();
+      }
+      if ( $action == 'execute' ) {
+        executeFilter($_REQUEST['Id']);
```

Comment:
I think this may have broke filters. When I save a filter, it doesn't send an ID across so this seems like it could cause a problem. See #2498

Reply:
not having an Id means that it acts like Save As.  Which should create a new Filter.

Reply:
Good point. And the action isn't execute in that case.